src/update-git-refs.ts

```diff
@@ -106,19 +133,25 @@ async function checkRefsAgainstGitHubAndModifyScalars(
       ref: trackable.trackMutableRef,
     });
 
-    if (trackable.ref === trackable.trackMutableRef) {
-      // The mutable ref was written down in ref too. We always want to replace
-      // that with the SHA (and if we do the path-based check below we won't,
-      // because they're the same). This is something that might happen when
-      // you're first adding an app (ie just writing the same thing twice and
-      // letting the automation "correct" it to a SHA).
-      trackable.refScalarTokenWriter.write(trackedRefCommitSHA);
-      continue;
-    }
-
-    if (trackable.ref === trackedRefCommitSHA) {
-      // The thing we would write is already in the file.
-      continue;
+    // If a trackable docker ref is defined we want to set it to that
+    // commit if the tree sha is the same as the tracked ref tree sha
+    // Otherwise we could "bounce back" to the docker ref if someone
+    // pushes to the tracked ref but doesn't change the tree sha.
+    if (!trackable.maybeDockerCommit) {
```

Comment:
I think this logic isn't quite right. Let's say trackMutableRef (or track) and ref are both set to `main`, and there is a docker commit, but the docker commit doesn't have the same tree SHA as main.  This whole section will get skipped, but then we'll hit the "unchanged" path and not rewrite main to the full commit SHA.

I think instead of wrapping this logic in this conditional we should drop it and incorporate it with the other logic changes later. The goal was to skip looking up tree SHAs (which should now be fast because we fetch them all at once anyway) in these two cases, but we can't really short-circuit in the "main points to given ref" anyway if our goal is to match docker. So might as well move farther down